### PR TITLE
55-update readme

### PR DESCRIPTION
**Pull request by AbhiramSureshkumar**
- Updated README.md to incorporate new updates for the following milestone with new schema and UML

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,19 +11,28 @@ Each product is identified by its online listing link and categorized by type. U
...
 ## Current State
-- Able to create a new user and add a review by making HTTP requests to the backend 
-- Implemented `User`, `Product`, and `Review` JPA entities, along with `UserController` and `ProductController`, made for RESTful API endpoints 
+- Able to create a new user and add a review by making HTTP requests to the backend
+- Implemented `User`, `Product`, and `Review` JPA entities, along with `UserController` and `ProductController`, made for RESTful API endpoints
 - CRUD operations for products and reviews, including creating a user and adding reviews for users
 - Validation and Error Handling: Unit tests for entities and controllers, input validation for URLs, null checks for entities and exception handling
- 
+- Added username uniqueness checks and related validation
+- Added RESTful API endpoints for:
```
Comment (hubertdang) [inline README.md]: Noy just API endpoints, we have the full features working dor this, right? Like, UI included

File: README.md [hunk 2]
```diff
@@ -10,20 +10,38 @@ Each product is identified by its online listing link and categorized by type. U
...
+  - Notifying a user login via email
+  - Environment variable management
+  - Updating project dependencies
+  - [Click this link for email service repository](https://github.com/AmileshN/SlayScaleEmailService)
+- Improved error messages and unified response structure
+- Added more robust unit tests for new controller paths, validation logic, and edge cases
+
+## Next Sprint Plan
+- Implement a sign-in form for existing users to login
+- Implement an SPA to support the client-side
```
Comment (AmileshN) [inline README.md]: @Aashna-Verma Didn't you mention something about cleaning up the code or smt for milestone 3?

**Review by AmileshN — APPROVED** [review]
